networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/peers/RateTracker.java

```diff
@@ -35,27 +42,94 @@ public RateTracker(
 
   // boundary: if a request comes in and remaining capacity is at least 1, then
   // they can have the objects they request otherwise they get none.
-  public synchronized long popObjectRequests(final long objectCount) {
+  public synchronized Optional<RequestApproval> popObjectRequests(final long objectCount) {
     pruneRequests();
-    if ((peerRateLimit - requestsWithinWindow) <= 0) {
-      return 0L;
+    final UInt64 currentTime = timeProvider.getTimeInSeconds();
+    if ((peerRateLimit - objectsWithinWindow) <= 0) {
+      return Optional.empty();
     }
+    objectsWithinWindow += objectCount;
+    resetRequestId(currentTime);
```

Comment:
I think we don't need this, we have pretty big margin to be sure we are not running out of numbers, especially as tracker is tied to a single connection

Reply:
Hmm, yes I see the rate tracker is initialised when `onConnect`. We could avoid resetting it

Reply:
Done